src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperResolver.cs

```diff
@@ -75,32 +74,21 @@ public DefaultTagHelperResolver(ErrorReporter errorReporter, Workspace workspace
                         nameof(GetTagHelpersAsync)),
                     exception);
             }
-        }
 
-        public override TagHelperResolutionResult GetTagHelpers(Compilation compilation)
-        {
-            var descriptors = new List<TagHelperDescriptor>();
-
-            var providers = new ITagHelperDescriptorProvider[]
-            {
-                new DefaultTagHelperDescriptorProvider() { DesignTime = true, },
-                new ViewComponentTagHelperDescriptorProvider(),
-            };
-
-            var results = new List<TagHelperDescriptor>();
-            var context = TagHelperDescriptorProviderContext.Create(results);
-            context.SetCompilation(compilation);
-
-            for (var i = 0; i < providers.Length; i++)
+            if (result == null)
             {
-                var provider = providers[i];
-                provider.Execute(context);
+                // Was unable to get tag helpers OOP, fallback to default behavior.
+                result = await _inProcessResolver.GetTagHelpersAsync(project, cancellationToken);
             }
 
-            var diagnostics = new List<RazorDiagnostic>();
-            var resolutionResult = new TagHelperResolutionResult(results, diagnostics);
+            return result;
+        }
+
+        public override TagHelperResolutionResult GetTagHelpers(Compilation compilation)
```

Comment:
We should look at this together and figure out if we still need both of these methods.

Reply:
We don't